src/Data/Array/Accelerate/AST.hs

```diff
@@ -1247,11 +1247,11 @@ liftMessage aR (Message _ fmt msg) =
       -- back to displaying in representation format
       fmtR :: ArraysR arrs' -> Q (TExp (arrs' -> String))
       fmtR TupRunit                         = [|| \() -> "()" ||]
-      fmtR (TupRsingle (ArrayR ShapeRz eR)) = [|| \as -> showElt $$(liftTypeR eR) $ linearIndexArray $$(liftTypeR eR) as 0 ||]
+      fmtR (TupRsingle (ArrayR ShapeRz eR)) = [|| \as -> showElt $$(liftTypeR eR) `id` linearIndexArray $$(liftTypeR eR) as 0 ||]
       fmtR (TupRsingle (ArrayR shR eR))     = [|| \as -> showArray (showsElt $$(liftTypeR eR)) (ArrayR $$(liftShapeR shR) $$(liftTypeR eR)) as ||]
       fmtR aR'                              = [|| \as -> showArrays $$(liftArraysR aR') as ||]
```

Comment:
It's kind of unfortunate that TemplateHaskell syntax itself is not only dependent on the currency in question, but also on large amounts of it, right?

Reply:
That might even be a larger problem!

Reply:
Though that would require a GHC plugin to fix it probably